### PR TITLE
[kms] Allow encryption via PEERDB_KMS_KEY_ID

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

**Review by serprex — APPROVED** [review]

**Comment by serprex** [comment]
Too bad https://github.com/smithy-lang/smithy-rs/pull/3683 is needed to not bring back hyper 0.14

File: flow/peerdbenv/env.go [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 	"encoding/json"
+	"fmt"
 	"os"
 	"reflect"
 	"strconv"
+	"strings"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kms"
```
Comment (iamKunalGupta) [inline flow/peerdbenv/env.go:14]: Need to switch to `aws-sdk-go-v2` for pod identity to work mostly

Comment (iskakaushik) [reply flow/peerdbenv/env.go]: done

File: flow/go.mod [hunk 2]
```diff
@@ -78,8 +78,10 @@ require (
 	github.com/DataDog/zstd v1.5.6 // indirect
 	github.com/JohnCGriffin/overflow v0.0.0-20211019200055-46fa312c352c // indirect
 	github.com/apache/arrow/go/v15 v15.0.2 // indirect
+	github.com/aws/aws-sdk-go v1.55.5 // indirect
```
Comment (serprex) [inline flow/go.mod:81]: how did sdk get added?

Comment (iskakaushik) [reply flow/go.mod]: removed it `go mod tidy`

**Pull request merged**